app/src/main/java/com/sun/hero_01/ui/class/ClassAdapter.kt

```diff
@@ -0,0 +1,93 @@
+package com.sun.hero_01.ui.`class`
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.hero_01.R
+import com.sun.hero_01.data.model.Hero
+import com.sun.hero_01.utils.Constant
+import com.sun.hero_01.utils.HeroDifficulty
+import com.sun.hero_01.utils.LoadImageBitmap
+import com.sun.hero_01.utils.OnItemRecyclerViewListener
+import kotlinx.android.synthetic.main.item_layout_hero.view.*
+
+class ClassAdapter(private val onItemClickListener: OnItemRecyclerViewListener<Hero>?) :
+    RecyclerView.Adapter<ClassAdapter.ViewHolder>() {
+
+    private val listHeroes = mutableListOf<Hero>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item_layout_hero, parent, false)
+        return ViewHolder(view, onItemClickListener)
+    }
+
+    override fun getItemCount() = listHeroes.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bindViewData(listHeroes[position])
+    }
+
+    fun updateData(heroes: MutableList<Hero>?) {
+        heroes?.let {
+            this.listHeroes.clear()
```

Comment:
Tại sao lại dùng this để tham chiếu, bỏ đi có được k ?

Reply:
Dạ để chỉ định đang gọi đến thuộc tính của class ạ. Nếu bỏ đi thì vẫn được ạ. This giúp chúng ta phân biệt khi có 2 biến giống nhau ( chẳng hạn như một thuộc tính của class và một thuộc tính hoặc một tham số trong một function chứa trong  class đó)